apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -155,6 +156,16 @@ internal SpannerCommand(string commandText, SpannerParameterCollection parameter
             }
         }
 
+        internal SpannerCommand(SpannerCommandTextBuilder commandTextBuilder, SpannerConnection connection, KeySet keySet, long limit, string indexName = null)
```

Comment:
If we move `limit` and `index` to the text builder, this is cleaner, also, I think we need to accept transactions here.

This constructor would be the same as the one that takes a command text builder, connection, transaction, an parameters, swapping that parameters for the key set only.

And at that point, we might even want to make it public.

(Adding this here because I can't add it in unmodified code)
And we should support partitions as well, because the Read API does. For partitions, we can make the `ExecutableCommand.GetReaderPartitionsAsync` distinguish between the Query and the Read to create the actual partitions and return the CommandPartition containing a Read or Sql requests (luckily all things there that we need to change are internal).

Reply:
I've moved the `limit` and `index` to the `ReadOptions` on the `SpannerCommandTextBuilder`, which cleans up the number of parameters in this constructor. I am however not sure that we should make this a public constructor that also accepts a transaction for the following reasons:
1. `KeySet` is required, which means that we cannot add it as an optional argument at the end of the arguments list (that is; it will be in a different position than the SpannerParameterCollection argument for the similar constructor)
2. `KeySet` is currently only supported for read commands (and soon for delete commands), which means that the constructor can only be used with `CommandTextBuilders` that are one of those types. I'm not sure that makes a better API than only having the factory methods on the `SpannerConnection` (the `CreateReadCommand` etc.)

I've made the constructor public for now so we can see what it looks like in that case, but I'm also open to making it internal again if we do not want this in the public API.

Reply:
Yep, agreed, I'm not sure either it should be public. Just a thought out of symmetry more than anything else. But agreed that maybe we just make it internal at first and we can always make it public later if we need to, which we cannot do the other way around.

Reply:
Partitioned read support has now been added as well.